test/runtests.jl

```diff
@@ -7,18 +7,26 @@ import Pkg
 # Make sure to not start with an outdated registry
 rm(joinpath(@__DIR__, "registries"); force = true, recursive = true)
 
-include("utils.jl")
-include("new.jl")
-include("pkg.jl")
-include("repl.jl")
-include("api.jl")
-include("registry.jl")
-include("subdir.jl")
-include("artifacts.jl")
-include("binaryplatforms.jl")
-include("platformengines.jl")
-include("sandbox.jl")
-include("resolve.jl")
+testfiles = [
+    "utils",
+    "new",
+    "pkg",
+    "repl",
+    "api",
+    "registry",
+    "subdir",
+    "artifacts",
+    "binaryplatforms",
+    "platformengines",
+    "sandbox",
+    "resolve",
+]
+original_project = Base.active_project()
+for testfile in testfiles
+    @info "Running test/$(testfile).jl"
+    include("$(testfile).jl")
+    Pkg.activate(original_project)
```

Comment:
This kind of hides the problem. Each test that does stuff with the project should ensure it goes back (by using e.g. isolate). If a test it not, it should be fixed. Otherwise, tests after it (in the same file) might be running with a weird state.

Reply:
Sounds good. Let me work in identifying the troublemaker tests, and wrapping them in `isolate() do ... end`